app/models/episode.rb

```diff
@@ -263,23 +323,17 @@ def media_status
   end
 
   def media_ready?
-    # if this episode has enclosures, media is ready if there is a complete one
-    if !enclosures.blank?
-      !!enclosure
-      # if this episode has contents, ready when each position is ready
-    elsif !all_contents.blank?
-      max_pos = all_contents.map(&:position).max
-      contents.size == max_pos
-      # if this episode has no audio, the media can't be ready, and `media?` will be false
+    if contents.blank?
+      ready_enclosure.present?
+    elsif segment_count.nil?
+      ready_contents.count == contents.maximum(:position)
```

Comment:
Is there a separate validation for 'holes' in the position sequence on either side here?

Reply:
Yeah, that's another validation we'll need to add at some point (which could potentially conflict with CMS).  Currently there is no tie/validation between segment_count and your contents.  That all lives on the CMS side.

I may need to think through if we can/should add all those validations sooner ... but that can be a separate issue from this.